EventFilter/CastorRawToDigi/src/ZdcUnpacker.cc

```diff
@@ -0,0 +1,520 @@
+#include "EventFilter/CastorRawToDigi/interface/ZdcUnpacker.h"
+#include "EventFilter/HcalRawToDigi/interface/HcalDCCHeader.h"
+#include "EventFilter/HcalRawToDigi/interface/HcalHTRData.h"
+#include "EventFilter/HcalRawToDigi/interface/HcalTTPUnpacker.h"
+#include "DataFormats/HcalDetId/interface/HcalOtherDetId.h"
+#include "DataFormats/HcalDigi/interface/HcalQIESample.h"
+#include "DataFormats/HcalDigi/interface/ZDCDataFrame.h"
+#include "DataFormats/HcalDigi/interface/HcalTriggerPrimitiveSample.h"
+#include "EventFilter/HcalRawToDigi/interface/AMC13Header.h"
+#include "EventFilter/HcalRawToDigi/interface/HcalDTCHeader.h"
+#include <iostream>
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <map>
+
+namespace ZdcUnpacker_impl {
+  template <class DigiClass>
+  const HcalQIESample* unpack(const HcalQIESample* startPoint, const HcalQIESample* limit, DigiClass& digi, int presamples, const CastorElectronicsId& eid, int startSample, int endSample, int expectedTime, const HcalHTRData& hhd) {
+    // set parameters
+    digi.setPresamples(presamples);
+    int fiber=startPoint->fiber();
+    int fiberchan=startPoint->fiberChan();
+    uint32_t zsmask=hhd.zsBunchMask()>>startSample;
+    digi.setZSInfo(hhd.isUnsuppressed(),hhd.wasMarkAndPassZS(fiber,fiberchan),zsmask);
+
+    //   digi.setReadoutIds(eid);
+    //   setReadoutIds is missing in  ZdcDataFrame class  digi.setReadoutIds(eid);
+    if (expectedTime>=0 && !hhd.isUnsuppressed()) {
+      // std::cout << hhd.getFibOrbMsgBCN(fiber) << " " << expectedTime << " fiber="<<fiber<< std::endl;
```

Comment:
please cleanup commented out code.
If you need these couts for debugging, you can convert them to LogDebug or LogTrace and include them in the code.